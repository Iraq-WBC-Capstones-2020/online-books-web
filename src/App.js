import React from 'react';
import './i18n';
import { HashRouter, Switch, Route, Redirect } from 'react-router-dom';
import { useSelector } from 'react-redux';
import Navbar from './Pages/nav';
import Footer from './components/Footer/Footer';
import AboutUsPage from './Pages/AboutUsPage/AboutUsPage';
import AuthorPage from './Pages/AuthorPage/AuthorPage';
import ProfilePage from './Pages/ProfilePage/ProfilePage';
import LoginPage from './Pages/loginPage/LoginPage';
import MainPage from './Pages/MainPage/MainPage';
import BookPage from './Pages/BookPage/BookPage';
import PaymentPage from './Pages/PaymentPage/PaymentPage';
import ContactUsPage from './Pages/ContactUsPage/ContactUsPage';
import SignUpPage from './Pages/SignUpPage/SignUpPage';

function App() {
  const user = useSelector((state) => state.user);

  return (
    <HashRouter>
      <Navbar />
      <Switch>
        <Route exact path="/book/:bookId" component={BookPage} />
        <Route exact path="/aboutus" component={AboutUsPage} />
        <Route exact path="/contact" render={ContactUsPage} />
        <Route exact path="/purchase">
          {user.isSigned ? <PaymentPage /> : <Redirect to="/login" />}
        </Route>
        <Route path="/author/:authorId" component={AuthorPage} />
        <Route exact path="/profile">
          {user.isSigned ? <ProfilePage /> : <Redirect to="/login" />}
        </Route>
        <Route exact path="/login">
          {!user.isSigned ? <LoginPage /> : <Redirect to="/profile" />}
        </Route>
        <Route exact path="/signup">
          {!user.isSigned ? <SignUpPage /> : <Redirect to="/profile" />}
        </Route>
        <Route path="/" component={MainPage} />
      </Switch>
      <Footer />
    </HashRouter>
  );
}

export default App;
