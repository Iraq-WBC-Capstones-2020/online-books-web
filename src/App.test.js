import React from 'react';
import App from './App';
import renderer from 'react-test-renderer';

test('Snapshot test App', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});
