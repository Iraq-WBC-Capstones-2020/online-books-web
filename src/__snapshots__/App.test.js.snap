// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test App 1`] = `
<h1>
  Online Book
</h1>
`;
